wpengine/faustjs
# test: End to end testing integration

This pull request adds [Cypress](https://www.cypress.io/) to the root of the project. In addition, the [wp-env](https://github.com/WordPress/gutenberg/tree/master/packages/env) package has been added that will allow end-2-end testing or local development.

There are currently only a few sample tests for the WPE Headless plugin settings page.

**Test**:
1. Pull down this branch `ryanmeier/add-end-2-end-testing`
2. Run `npm install` within the root directory of the project.
3. Ensure Docker is running.
4. Run `npm run wp:start` to start the WordPress development server (docker). The initial start will download and setup the site.
    - The development site will be at `http://localhost:8888` with the testing site at `http://localhost:8889`.
5. Run `npm run cypress:open` to open the Cypress UI, this will allow you to visually see the tests run.
    - You can also run `npm run cypress:run` to just run the tests without the UI (will be used in CircleCI).
6. Run `npm wp:stop` to stop the container.

**Commands**:
- `npm run wp-env` Run `wp-env` [commands](https://github.com/WordPress/gutenberg/tree/master/packages/env#usage). 
- `npm run wp:start` Start the WordPress development environment.
- `npm run wp:stop` Stop the WordPress development environment.
- `npm run wp:destroy` Destoy the WordPress development environment.
- `npm run cypress:run` Run the end-2-end tests.
- `npm run cypress:open` Open the Cypress UI to visually run tests.

**Docs**:
- [Cypress Docs](https://docs.cypress.io/guides/overview/why-cypress.html#In-a-nutshell)
- [Mocha Docs](https://mochajs.org/)
- [wp-env Docs](https://github.com/WordPress/gutenberg/tree/master/packages/components/src)
